File: desktop/src/grade_level.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize, Default)]
pub enum GradeLevel {
    #[default]
    PreK,
    Kindergarten,
    First,
    Second,
    Third,
    Fourth,
    Fifth,
    Sixth,
}

impl GradeLevel {
    #[allow(dead_code)]
    pub fn all() -> Vec<GradeLevel> {
        vec![
            GradeLevel::PreK,
            GradeLevel::Kindergarten,
            GradeLevel::First,
            GradeLevel::Second,
            GradeLevel::Third,
            GradeLevel::Fourth,
            GradeLevel::Fifth,
            GradeLevel::Sixth,
        ]
    }

    pub fn display_name(&self) -> &str {
        match self {
            GradeLevel::PreK => "Pre-K",
            GradeLevel::Kindergarten => "Kindergarten",
            GradeLevel::First => "1st Grade",
            GradeLevel::Second => "2nd Grade",
            GradeLevel::Third => "3rd Grade",
            GradeLevel::Fourth => "4th Grade",
            GradeLevel::Fifth => "5th Grade",
            GradeLevel::Sixth => "6th Grade",
        }
    }

    #[allow(dead_code)]
    pub fn short_name(&self) -> &str {
        match self {
            GradeLevel::PreK => "Pre-K",
            GradeLevel::Kindergarten => "K",
            GradeLevel::First => "1st",
            GradeLevel::Second => "2nd",
            GradeLevel::Third => "3rd",
            GradeLevel::Fourth => "4th",
            GradeLevel::Fifth => "5th",
            GradeLevel::Sixth => "6th",
        }
    }

    pub fn next_level(&self) -> Option<GradeLevel> {
        match self {
            GradeLevel::PreK => Some(GradeLevel::Kindergarten),
            GradeLevel::Kindergarten => Some(GradeLevel::First),
            GradeLevel::First => Some(GradeLevel::Second),
            GradeLevel::Second => Some(GradeLevel::Third),
            GradeLevel::Third => Some(GradeLevel::Fourth),
            GradeLevel::Fourth => Some(GradeLevel::Fifth),
            GradeLevel::Fifth => Some(GradeLevel::Sixth),
            GradeLevel::Sixth => None,
        }
    }

    pub fn previous_level(&self) -> Option<GradeLevel> {
        match self {
            GradeLevel::PreK => None,
            GradeLevel::Kindergarten => Some(GradeLevel::PreK),
            GradeLevel::First => Some(GradeLevel::Kindergarten),
            GradeLevel::Second => Some(GradeLevel::First),
            GradeLevel::Third => Some(GradeLevel::Second),
            GradeLevel::Fourth => Some(GradeLevel::Third),
            GradeLevel::Fifth => Some(GradeLevel::Fourth),
            GradeLevel::Sixth => Some(GradeLevel::Fifth),
        }
    }
}

File: ios/WordsWithToddlers/WordsWithToddlers/Models/GradeLevel.swift
import Foundation

enum GradeLevel: String, Codable, CaseIterable {
    case preK = "PreK"
    case kindergarten = "Kindergarten"
    case first = "First"
    case second = "Second"
    case third = "Third"
    case fourth = "Fourth"
    case fifth = "Fifth"
    case sixth = "Sixth"

    var displayName: String {
        switch self {
        case .preK: return "Pre-K"
        case .kindergarten: return "Kindergarten"
        case .first: return "1st Grade"
        case .second: return "2nd Grade"
        case .third: return "3rd Grade"
        case .fourth: return "4th Grade"
        case .fifth: return "5th Grade"
        case .sixth: return "6th Grade"
        }
    }

    var shortName: String {
        switch self {
        case .preK: return "Pre-K"
        case .kindergarten: return "K"
        case .first: return "1st"
        case .second: return "2nd"
        case .third: return "3rd"
        case .fourth: return "4th"
        case .fifth: return "5th"
        case .sixth: return "6th"
        }
    }

    func nextLevel() -> GradeLevel? {
        let levels = GradeLevel.allCases
        guard let currentIndex = levels.firstIndex(of: self),
              currentIndex < levels.count - 1 else {
            return nil
        }
        return levels[currentIndex + 1]
    }

    func previousLevel() -> GradeLevel? {
        let levels = GradeLevel.allCases
        guard let currentIndex = levels.firstIndex(of: self),
              currentIndex > 0 else {
            return nil
        }
        return levels[currentIndex - 1]
    }
}
